pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -680,9 +685,10 @@ func (c *Controller) serviceInstancesFromWorkloadInstances(svc *model.Service, r
 	selector := labels.Instance(svc.Attributes.LabelSelectors)
 
 	// Get the service port name and target port so that we can construct the service instance
-	k8sService, err := c.serviceLister.Services(svc.Attributes.Namespace).Get(svc.Attributes.Name)
-	// We did not find the k8s service. We cannot get the targetPort
-	if err != nil {
+	k8sService := &v1.Service{}
+	key := types.NamespacedName{Name: svc.Attributes.Name, Namespace: svc.Attributes.Namespace}
```

Comment:
Any reason to use a struct vs just passing 2 strings?

Reply:
Not really. Just another bit that is similar to the controller-runtime API. I don't have an opinion one way or the other on this really.

Reply:
I actually prefer this personally. There are a lot of places where the order of name/namespace is reversed ... which is pretty error-prone.